pkg/segment/aggregations/segaggs.go

```diff
@@ -152,9 +152,19 @@ func performAggOnResult(nodeResult *structs.NodeResult, agg *structs.QueryAggreg
 				return fmt.Errorf("performAggOnResult: %v", err)
 			}
 		}
+
+		if agg.OutputTransforms.MaxRows > 0 {
+			err := performMaxRows(nodeResult, agg, agg.OutputTransforms.MaxRows, recs)
+
+			if err != nil {
+				return fmt.Errorf("performAggOnResult: %v", err)
+			}
+		}
 	case structs.GroupByType:
 		nodeResult.PerformAggsOnRecs = true
 		nodeResult.RecsAggsType = structs.GroupByType
+		nodeResult.GroupByCols = agg.GroupByRequest.GroupByColumns
+		nodeResult.GroupByRequest = agg.GroupByRequest
```

Comment:
- I have added these back here, otherwise for some of the `stats` aggs, these values are not set.
- For example, the flow of extracting rex columns followed by stats, then the aggs at the `PostQueryBucketCleaning` will first have `rex-col aggs` and once that is processed, it will go for the next agg which is the `stats` inside the for loop at `PostQueryBucketCleaning` function without setting these values. 

- When I executed the below command, it `paniced due to nil reference` at `searchaggs.go:PerformAggsOnRecs` 
```SPL
* | rex field=app_version "(?<major>\d+)\.(?<minor>\d).*" | stats count by major
```